cli/kaos_cli/services/tests/test_state_service.py

```diff
@@ -0,0 +1,228 @@
+import os
+import time
+from parameterized import parameterized
+from configobj import ConfigObj
+
+from unittest import TestCase
+
+from kaos_cli.services.state_service import StateService
+from kaos_cli.constants import CONFIG_PATH, KAOS_STATE_DIR
+
+
+def state_service_test_inputs():
+    return [
+        (
+            'context1', 'param1'
+        ),
+        (
+            'context2', 'param2'
+        ),
+        (
+            '', 'param3'
+        ),
+        (
+            'context4', ''
+        ),
+        (
+            '', ''
+        ),
+        (
+            {}, 'param6'
+        ),
+        (
+            'context7', {}
+        ),
+        (
+            {}, {}
+        ),
+        (
+            '', {}
+        ),
+        (
+            {}, ''
+        ),
+        (
+            None, 'param11'
+        ),
+        (
+            'context11', None
+        ),
+        (
+            None, None
+        ),
+        (
+            {}, None
+        ),
+        (
+            None, {}
+        ),
+        (
+            '', None
+        ),
+        (
+            None, ''
+        ),
+        (
+            1, 'param18'
+        ),
+        (
+            'context', 2
+        ),
+        (
+            3, {}
+        ),
+        (
+            {}, 4
+        ),
+        (
+            5, ''
+        ),
+        (
+            '', 6
+        ),
+        (
+            7, None
+        ),
+        (
+            None, 8
+        ),
+
+    ]
```

Comment:
What do all these values mean? Please document this method. Also, would it be possible to use a DataClass or a NamedTuple to hold these values in a meaningful named type?

Reply:
These are the different parameter combinations that are used to test the behaviour of each unit test. Please refer the following link for further info: https://pypi.org/project/parameterized/

In this particular case, these values are used to test various context-parameter combinations. These methods are documented appropriately as suggested.

Unfortunately, these parameter combinations can only be passed as tuples. However, the ```test_state_service``` now uses a ```@parameterized_class``` instead of the approach used earlier, which considerably reduces verbosity. Please check to the above link that illustrates both these approaches provided by the library.